src/main/java/org/folio/service/holdings/HoldingsServiceImpl.java

```diff
@@ -38,21 +42,30 @@
   private long delay;
   private int retryCount;
   private HoldingsRepository holdingsRepository;
+  private HoldingsStatusRepository holdingsStatusRepository;
 
   @Autowired
   public HoldingsServiceImpl(Vertx vertx, HoldingsRepository holdingsRepository,
+                             HoldingsStatusRepository holdingsStatusRepository,
                              @Value("${holdings.status.check.delay}") long delay,
                              @Value("${holdings.status.retry.count}") int retryCount) {
     this.vertx = vertx;
     this.holdingsRepository = holdingsRepository;
+    this.holdingsStatusRepository = holdingsStatusRepository;
     this.delay = delay;
     this.retryCount = retryCount;
   }
 
-  public CompletableFuture<Void> loadHoldings(RMAPITemplateContext context, String tenantId) {
+  public CompletableFuture<Void> loadHoldings(RMAPITemplateContext context) {
+    final String tenantId = context.getOkapiData().getTenant();
     return populateHoldings(context)
+      .thenAccept(o -> updateStatus(getStatusPopulatingStagingArea(), tenantId))
```

Comment:
let's update the status inside populateHoldings method() to keep the consistency with loadHoldings()